internal/service/odb/cloud_exadata_infrastructure.go

```diff
@@ -0,0 +1,848 @@
+			},
+			"next_maintenance_run_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The OCID of the next maintenance run for the Exadata infrastructure",
+			},
+			"ocid": schema.StringAttribute{
+				Computed:    true,
+				Description: "The OCID of the Exadata infrastructure",
+			},
+			"oci_resource_anchor_name": schema.StringAttribute{
+				Computed:    true,
+				Description: "The name of the OCI resource anchor for the Exadata infrastructure",
+			},
+			"oci_url": schema.StringAttribute{
+				Computed:    true,
+				Description: "The HTTPS link to the Exadata infrastructure in OCI",
+			},
+			"percent_progress": schema.Float64Attribute{
+				Computed:    true,
+				Description: "The amount of progress made on the current operation on the Exadata infrastructure, expressed as a percentage",
+			},
+			"shape": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Description: "The model name of the Exadata infrastructure. Changing this will force terraform to create new resource",
+			},
+			"status": schema.StringAttribute{
+				CustomType:  statusType,
+				Computed:    true,
+				Description: "The current status of the Exadata infrastructure",
+			},
+			"status_reason": schema.StringAttribute{
+				Computed:    true,
+				Description: "Additional information about the status of the Exadata infrastructure",
+			},
+			"storage_count": schema.Int32Attribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.Int32{
+					int32planmodifier.RequiresReplace(),
+					int32planmodifier.UseStateForUnknown(),
+				},
+				Description: "TThe number of storage servers that are activated for the Exadata infrastructure",
+			},
+			"storage_server_version": schema.StringAttribute{
+				Computed:    true,
+				Description: "The software version of the storage servers on the Exadata infrastructure.",
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			"total_storage_size_in_gbs": schema.Int32Attribute{
+				Computed: true,
+			},
+			"created_at": schema.StringAttribute{
+				Computed:    true,
+				Description: "The time when the Exadata infrastructure was created",
+			},
+			"compute_model": schema.StringAttribute{
+				CustomType: computeModelType,
+				Computed:   true,
+				Description: fmt.Sprint("The OCI model compute model used when you create or clone an\n " +
+					" instance: ECPU or OCPU. An ECPU is an abstracted measure of\n " +
+					"compute resources. ECPUs are based on the number of cores\n " +
+					"elastically allocated from a pool of compute and storage servers.\n " +
+					" An OCPU is a legacy physical measure of compute resources. OCPUs\n " +
+					"are based on the physical core of a processor with\n " +
+					" hyper-threading enabled."),
+			},
+			"maintenance_window": schema.ObjectAttribute{
```

Comment:
for backwards compatibility with Terraform, and consistence across the provider, `ObjectAttribute` should be created as `ListNestedBlock` in the schema with a size max of one. A good example can be found [here](https://github.com/hashicorp/terraform-provider-aws/blob/2dc892ec0a9c1f760dd021433911690ffd768b57/internal/service/osis/pipeline.go#L166)